doc/source/dev/data_types.md

```diff
@@ -0,0 +1,311 @@
+        ...
+```
+**Note:** If you do not wish to add extended functionality to for a new datatype, but simply want to restrict the output of a set of tools to be used in another set of tools, you can add the flag `subclass="True"` to the datatype definition line. 
+
+Example:
+
+```xml
+<datatype extension="my_tabular_subclass" type="galaxy.datatypes.tabular:Tabular" subclass="True"/>
+```
+
+### Step 2: Sniffer
+
+Galaxy tools are configured to automatically set the data type of an output dataset. However, in some scenarios, Galaxy will attempt to determine the data type of a file using a sniffer (e.g., uploading a file from a local disk with 'Auto-detect' selected in the File Format select list). The order in which Galaxy attempts to determine data types is critical because some formats are much more loosely defined than others. The order in which the sniffer for each data type is applied to the file should be most rigidly defined formats first followed by less and less rigidly defined formats, with the most loosely defined format last, and then a default format associated with the file if none of the data type sniffers were successful. The order in which data type sniffers are applied to files is implicit in the `<sniffers>` tag set section of the `datatypes_conf.xml` file. We'll assume that the format of our Foobar data type is fairly rigidly defined, so it can be placed closer to the start of the sniff order.
+```xml
+<sniffers>
+    <sniffer type="galaxy.datatypes.sequence:Maf"/>
+    <sniffer type="galaxy.datatypes.sequence:Lav"/>
+    <sniffer type="galaxy.datatypes.tabular:Foobar"/>
+```
+
+
+### Step 3: Data Type Class
+
+We'll now create a new code file, `galaxy.datatypes.foobar.py`, that will contain the `Foobar` class (in this example we could simply add the `Foobar` class to the `galaxy.datatypes.tabular.py` code file). Keep in mind that your new data type class should be placed in a file that is appropriate (based on it's superclass), and that the file will need to be imported by `lib/galaxy/datatypes/registry.py`. You will need to include a `file_ext` attribute to your class and create any necessary functions to override the functions in your new data type's superclass (in our example, the `galaxy.datatypes.tabular.Tabular` class). In our example below, we have set our class's `file_ext` attribute to "foo" and we have overridden the `__init__()`, `init_meta()` and `sniff()` functions. It is important to override functions (especially the meta data and sniff functions) if the attributes of your new class differ from those of it's superclass. Note: sniff functions are not required to be included in new data type classes, but if the sniff function is missing, Galaxy will end up associating the default data type and file extension (`Text` and `txt` in our example) with the file. For binary files, the default would be `Data` and `data`.
```

Comment:
In the `datatypes_conf.xml` example above you use `galaxy.datatypes.tabular:Foobar`, so you need to also add `Foobar` to `lib/galaxy/datatypes/tabular.py`, not a new file.

```suggestion
We'll now add the `Foobar` class to `lib/galaxy/datatypes/tabular.py`. Keep in mind that your new data type class should be placed in a file that is appropriate (based on its superclass), and that the file will need to be imported by `lib/galaxy/datatypes/registry.py`. You will need to include a `file_ext` attribute to your class and create any necessary functions to override the functions in your new data type's superclass (in our example, the `galaxy.datatypes.tabular.Tabular` class).
In our example below, we have set our class's `file_ext` attribute to "foo" and we have overridden the `__init__()`, `init_meta()` and `sniff()` functions. It is important to override functions (especially the meta data and sniff functions) if the attributes of your new class differ from those of its superclass. Note: sniff functions are not required to be included in new data type classes, but if the sniff function is missing, Galaxy will call the superclass method.
```